Update KNX documentation for IQS (#39020)
home-assistant/home-assistant.github.io: source/_integrations/knx.markdown

ha_integration_type: hub
---

> The [KNX](https://www.knx.org) integration connects Home Assistant to your KNX installation, allowing you to control KNX devices, act on telegrams and forward states from other Home Assistant entities to your KNX bus.

Reviewer: ```suggestion
The [KNX](https://www.knx.org) integration enables Home Assistant to control KNX devices, process telegrams, and sync states with your KNX bus. A local KNX/IP interface or router is required for the connection.
```

Author: I like mine better 😛 changed it slightly in [c7e353b](https://github.com/home-assistant/home-assistant.io/pull/39020/commits/c7e353bec044ac83cc57da4d23311f202d60cd1b)

> This integration requires a local KNX/IP interface or router to establish the connection between Home Assistant and your KNX bus.

Reviewer: ```suggestion
```

Author: I prefer to have a separate paragraph to separate general use case of the integration and the hardware requirements.

{% include integrations/config_flow.md %}

### KNX Connection

Select the connection type to your KNX bus. The integration supports the following connection methods:

- 'Automatic' performs a gateway scan on start, to find a KNX IP interface. It will connect via a tunnel. This option is not available when a gateway scan at setup initiation was not successful.
- 'Tunneling' will connect to a specific KNX IP interface over a tunnel.
- 'Routing' will use Multicast to communicate with KNX IP routers.

For more information about KNX connection types see [Connection](#connection).

> ### Tunnelling specific settings

Reviewer: ```suggestion
### Tunneling specific settings
```

Author: Oh dear... we are currently running a mixed bag of `Tunneling` and `Tunnelling` everywhere (xknx, HA strings.json, config flow, constant and variable names, etc.). 
KNXA seems to prefer double-l version.
<img width="534" alt="Bildschirmfoto 2025-05-14 um 15 19 50" src="https://github.com/user-attachments/assets/aac0c0e9-09b1-4e9d-be55-26d31a128fcb" />

Author: Seems to be US vs GB English. Since HA prefers US I think you are correct.

> You can select a specific tunnel device if multiple were found during the setup. In case no device was discovered you can configure connection information manually.

Reviewer 2: ```suggestion
You can select a specific tunnel device if multiple were found during the setup. In case no device was discovered, you can configure connection information manually.
```

{% configuration_basic %}
KNX tunneling type:

>   description: "`UDP`, `TCP` or `Secure Tunnelling`"

Reviewer: ```suggestion
  description: "`UDP`, `TCP` or `Secure Tunneling`"
```

Host:
  description: "IP address or hostname of the KNX/IP tunneling device."
Port:
  description: "Port used by the KNX/IP tunneling device."
Route back / NAT mode:
  description: "Enable if your KNXnet/IP tunneling server is behind NAT. Only applies for UDP connections."
Local IP interface:
  description: "Local IP or interface name used for the connection from Home Assistant. Leave blank to use auto-discovery."
{% endconfiguration_basic %}

#### Tunnel endpoint

> Select the tunnel endpoint used for the connection. This step is only available for `TCP` or `Secure Tunnelling` connection types.

Reviewer: ```suggestion
Select the tunnel endpoint used for the connection. This step is only available for `TCP` or `Secure Tunneling` connection types.
```

Multicast port:
  description: "Multicast port used by your installation. Default is `3671`"
Local IP interface:
  description: "Local IP or interface name used for the connection from Home Assistant. Leave blank to use auto-discovery."
{% endconfiguration_basic %}

### KNX IP Secure specific settings

See [Connection](#connection) on how to get the files or keys needed for this configuration step.

{% include integrations/option_flow.md %}

### Configure KNX interface

Reconfigure your connection settings. See above for more information.

### Communication settings

{% configuration_basic %}
State updater:
  description: "Sets the default behavior for periodically reading state addresses from the KNX Bus."
Rate limit:
  description: "Maximum outgoing telegrams per second. `0` to disable limit - which is recommended."
Telegram history limit:
  description: "Number of Telegrams to keep in memory for the KNX panels group monitor."
{% endconfiguration_basic %}

### Import KNX Keyring

Provide a (new) keyring file to be used by the integration. See [KNX Secure](#knx-secure) on how to get this file.

## Basic configuration

> In order to make use of the various platforms offered by the KNX integration, you will need to set them up via the KNX panel or add the according configuration yaml to your {% term "`configuration.yaml`" %}. See [Splitting up the configuration](/docs/configuration/splitting_configuration/) if you like to arrange yaml parts in dedicated files.

Reviewer: ```suggestion
In order to make use of the various platforms offered by the KNX integration, you will need to set them up via the KNX panel or add the corresponding configuration yaml to your {% term "`configuration.yaml`" %}. See [Splitting up the configuration](/docs/configuration/splitting_configuration/) if you like to arrange yaml parts in dedicated files.
```

> In order to make use of the various platforms offered by the KNX integration, you will need to set them up via the KNX panel or add the corresponding configuration yaml to your {% term "`configuration.yaml`" %}. See [Splitting up the configuration](/docs/configuration/splitting_configuration/) if you like to arrange yaml parts in dedicated files.

Reviewer 2: ```suggestion
In order to make use of the various platforms offered by the KNX integration, you will need to set them up via the KNX panel or add the corresponding configuration yaml to your {% term "`configuration.yaml`" %}. See [Splitting up the configuration](/docs/configuration/splitting_configuration/) if you like to arrange YAML parts in dedicated files.
```

  default: None
{% endconfiguration %}

## Known limitations

> - The integration aims to be compatible with a wide variety of KNX devices from different manufacturers and eras. However, there are some devices that use non-standard <abbr title="data point type">DPT</abbr> or use telegrams in a proprietary way. In these cases, you might not be able to configure entities directly through this integration. However, you may still use [Template entities](integrations/template/) with the [KNX telegram trigger](#telegram-trigger) to work around this.

Reviewer 2: ```suggestion
- The integration aims to be compatible with a wide variety of KNX devices from different manufacturers and eras. However, there are some devices that use non-standard <abbr title="data point type">DPT</abbr> or use telegrams in a proprietary way. In these cases, you might not be able to configure {% term entities %} directly through this integration. However, you may still use [Template entities](integrations/template/) with the [KNX telegram trigger](#telegram-trigger) to work around this.
```

> - The integration aims to be compatible with a wide variety of KNX devices from different manufacturers and eras. However, there are some devices that use non-standard <abbr title="data point type">DPT</abbr> or use telegrams in a proprietary way. In these cases, you might not be able to configure entities directly through this integration. However, you may still be able use [Template entities](integrations/template/) with [KNX telegram trigger](#telegram-trigger) to work around this.

Reviewer: ```suggestion
- The integration aims to be compatible with a wide variety of KNX devices from different manufacturers and eras. However, there are some devices that use non-standard <abbr title="data point type">DPT</abbr> or use telegrams in a proprietary way. In these cases, you might not be able to configure entities directly through this integration. However, you may still use [Template entities](integrations/template/) with the [KNX telegram trigger](#telegram-trigger) to work around this.
```
